test/browser-pool.test.ts

```diff
@@ -569,6 +570,210 @@ describe('BrowserPool', () => {
+                        };
+                    });
+                    // @ts-expect-error - I don't care that it can be undefined
+
+                    const { fingerprint } = browserController.launchContext;
+
+                    expect(data.hardwareConcurrency).toBe(fingerprint?.navigator.hardwareConcurrency);
+                    expect(data.userAgent).toBe(fingerprint?.userAgent);
+
+                    await page.close();
+                    await browserPoolWithFP.destroy();
+                });
+                describe('caching', () => {
+                    const commonOptions = {
+                        browserPlugins: [new PlaywrightPlugin(
+                            playwright.chromium,
+                            {
+                                useIncognitoPages: true,
+                            },
+                        )],
+                    };
+                    // @ts-expect-error - dont know how to fix this
+                    let browserPoolCache: BrowserPool;
+                    afterEach(async () => {
+                        await browserPoolCache.destroy();
+                    });
+                    test('should use fingerprint per proxy by default', async () => {
+                        browserPoolCache = new BrowserPool({
+                            ...commonOptions,
+                            useFingerprints: true,
+                        });
+
+                        expect(browserPoolCache.fingerprintCache).toBeDefined();
+                    });
+
+                    test('should turn off cache', async () => {
+                        browserPoolCache = new BrowserPool({
+                            ...commonOptions,
+                            useFingerprints: true,
+                            fingerprintsOptions: {
+                                useFingerprintPerProxyCache: false,
+                            },
+                        });
+
+                        expect(browserPoolCache.fingerprintCache).toBeUndefined();
+                    });
+
+                    test('should limit cache size', async () => {
+                        browserPoolCache = new BrowserPool({
+                            ...commonOptions,
+                            useFingerprints: true,
+                            fingerprintsOptions: {
+                                fingerprintPerProxyCacheSize: 1,
+                            },
+                        });
+
+                        expect(browserPoolCache.fingerprintCache.cache.maxSize).toBe(1);
+                    });
+
+                    test('should cache fingerprints', async () => {
+                        browserPoolCache = new BrowserPool({
+                            ...commonOptions,
+                            useFingerprints: true,
+                            preLaunchHooks: [
+                                (_pageId, launchContext) => {
+                                    // @ts-expect-error WTF
```

Comment:
another issue caused by the generics, if you would specify them explicitly it would work, but this way the launch context resolves to `never`

keep in this way for now